eth/handler.go

```diff
@@ -305,51 +302,33 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		var (
 			hash   common.Hash
 			bytes  common.StorageSize
-			hashes []common.Hash
 			blocks []*types.Block
 		)
-		for {
+		for len(blocks) < downloader.MaxBlockFetch && bytes < softResponseLimit {
```

Comment:
MaxBlockFetch sounds like a inclusive bound, if it's not the comment where it's defined needs an update.